tests/subsys/zbus/unittests/src/main.c

```diff
@@ -760,4 +770,34 @@ ZTEST(basic, test_specification_based__zbus_sub_wait_msg)
 	irq_offload(isr_sub_wait_msg, NULL);
 }
 
+#if defined(CONFIG_ZBUS_PRIORITY_BOOST)
+static void isr_obs_attach_detach(const void *operation)
+{
+	zassert_equal(-EFAULT, zbus_obs_attach_to_thread(NULL), NULL);
+	zassert_equal(-EFAULT, zbus_obs_attach_to_thread(&foo_sub), NULL);
+	zassert_equal(-EFAULT, zbus_obs_attach_to_thread(&invalid_sub), NULL);
+
+	zassert_equal(-EFAULT, zbus_obs_detach_from_thread(NULL), NULL);
+	zassert_equal(-EFAULT, zbus_obs_detach_from_thread(&foo_sub), NULL);
+	zassert_equal(-EFAULT, zbus_obs_detach_from_thread(&invalid_sub), NULL);
+}
+
+ZTEST(basic, test_specification_based__zbus_obs_attach_detach)
+{
+	zassert_equal(-EFAULT, zbus_obs_attach_to_thread(NULL), NULL);
+	zassert_equal(0, zbus_obs_attach_to_thread(&foo_sub), NULL);
+	zassert_equal(0, zbus_obs_detach_from_thread(&foo_sub), NULL);
+	zassert_equal(0, zbus_obs_attach_to_thread(&invalid_sub), NULL);
+	zassert_equal(0, zbus_obs_detach_from_thread(&invalid_sub), NULL);
+	zassert_equal(-EFAULT, zbus_obs_detach_from_thread(NULL), NULL);
+
+	irq_offload(isr_obs_attach_detach, NULL);
+}
+#else
+ZTEST(basic, test_specification_based__zbus_obs_attach_detach)
+{
+	ztest_test_skip();
+}
+
```

Comment:
nit: drop, symmetry...

Reply:
I am following the documentation. Please change that to avoid confusion: https://docs.zephyrproject.org/latest/develop/test/ztest.html#skipping-tests

Reply:
I'm talking about the empty line, you have none separating the #if and #else but you left one on the #endif

Reply:
the doc has a good example though can you skip with `Z_TEST_SKIP_IFDEF` rather than building this out?

Reply:
I would like to skip it if it is not defined. Should I "not" that?

Reply:
of course, can you use `Z_TEST_SKIP_IFNDEF`? juts keeps things a bit neater and make sure the whole code gets compiled regarless of the config, you obviously can't do that if something you use in the test is in a conditional itself but I don't think it's the case here? If it's the case then fine as it is just drop the blank line. :-)

Same logic as #ifdef versus IS_ENABLED.

Reply:
I did not know there was such a thing. I will use that. Thanks.

Reply:
yeah me neither until you linked that doc :-)

Reply:
I see why I needed the ifdef style. The `zbus_obs_attach_to_thread` only exists when the priority boost is enabled.

Reply:
ok then fine as it is, just drop that empty line

Reply:
well I would only ifdef/endif on the content and not duplicate the `ZTEST(basic, test_specification_based__zbus_obs_attach_detach)` but up to you